api-service/crowdsource-webapp/src/app/answers-distribution-chart/answers-distribution-chart.component.ts

```diff
@@ -0,0 +1,123 @@
+import { AfterViewInit, Component, ElementRef, Input, ViewChild, ViewEncapsulation } from '@angular/core';
+import { ActivatedRoute, Params, ParamMap, Router } from '@angular/router';
+import {
+  AnswerSummary,
+  CrowdsourceApiService,
+} from '../crowdsource-api.service';
+import { RelativeToxicityAnswer, ToxicityOption } from '../relative-toxicity-job/relative-toxicity-job.component';
+import * as d3 from 'd3';
+
+interface QuestionsToAnswersMap {
+  [questionId: string]: AnswerSummary<RelativeToxicityAnswer>[];
+}
+
+const TOXICITY_OPTION_MAP = {};
+TOXICITY_OPTION_MAP[ToxicityOption.MUCH_MORE_TOXIC] = 0;
+TOXICITY_OPTION_MAP[ToxicityOption.MORE_TOXIC] = 1;
+TOXICITY_OPTION_MAP[ToxicityOption.SLIGHTLY_MORE_TOXIC] = 2;
+TOXICITY_OPTION_MAP[ToxicityOption.ABOUT_THE_SAME] = 3;
+TOXICITY_OPTION_MAP[ToxicityOption.SLIGHTLY_LESS_TOXIC] = 4;
+TOXICITY_OPTION_MAP[ToxicityOption.LESS_TOXIC] = 5;
+TOXICITY_OPTION_MAP[ToxicityOption.MUCH_LESS_TOXIC] = 6;
+
+const BAR_COLORS_MAP = {};
+BAR_COLORS_MAP[ToxicityOption.MUCH_MORE_TOXIC] = '#B71C1C';
+BAR_COLORS_MAP[ToxicityOption.MORE_TOXIC] = '#D32F2F';
+BAR_COLORS_MAP[ToxicityOption.SLIGHTLY_MORE_TOXIC] = '#EF5350';
+BAR_COLORS_MAP[ToxicityOption.ABOUT_THE_SAME] = '#FBC02D';
+BAR_COLORS_MAP[ToxicityOption.SLIGHTLY_LESS_TOXIC] = '#43A047';
+BAR_COLORS_MAP[ToxicityOption.LESS_TOXIC] = '#388E3C';
+BAR_COLORS_MAP[ToxicityOption.MUCH_LESS_TOXIC] = '#2E7D32';
+
+/** Describes a bar in the graph. */
+interface ToxicityOptionSummary {
+  bucketName: ToxicityOption;
+  count: number;
+}
+
+/**
+ */
+@Component({
+  selector: 'app-answers-distribution-chart',
+  templateUrl: './answers-distribution-chart.component.html',
+  styleUrls: ['./answers-distribution-chart.component.css'],
+  encapsulation: ViewEncapsulation.None,
+})
+export class AnswersDistributionChartComponent implements AfterViewInit {
+  @Input() answers: AnswerSummary<RelativeToxicityAnswer>[] = [];
+  @ViewChild('chart') chart: ElementRef;
+
+  constructor(
+    private route: ActivatedRoute,
+    private crowdSourceApiService: CrowdsourceApiService) { }
+
+  ngAfterViewInit(): void {
+    this.renderGraph();
+  }
+
+  getGraphData(): ToxicityOptionSummary[] {
```

Comment:
This isn't really a "getter", but more of a prepare data as it produces intermediate data. maybe better to call this "prepareGraphData()"

Reply:
Done.